### ISSUE TITLE
We log entirely too frequently from `Terminal::CursorLineFeed`

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/terminal#10882]
This is a follow-up for an internal bug. There's two pieces here:

* [ ] (**A**) Change the `CATCH_LOG_RETURN_FALSE` to a newly authored `CATCH_LOG_RETURN_FALSE_EXPECTED`. We're expecting this exception, so make it less noisy. This will fix MSFT:33310649
* [ ] (**B**) Get to the bottom of why `Terminal::CursorLineFeed` is throwing so often. It's an `E_ABORT` from `rectangle.h:840`, though that line of code is nothing in main, so we might need to look at what it is in 1.9. Presumably something in `_AdjustCursorPosition` throws, though impossible to say what exactly.

**Comment by zadjii-msft** [comment]
Huh well, I slapped a `DebugBreak()` in the exception handler in `Terminal::CursorLineFeed` and tried printing 1000000000 lines of text to see if that triggers the exception. No luck. 

Tried with a 9000001 size buffer (which would be clamped to `SHORT_MAX`) - no luck.

Tried with `cmatrix` - again, no luck. 

Resizing wildly also didn't seem to trigger the debugbreak. So I'm a little lost as to what would cause this one. Someone would actually need to find where we're doing the `til::rectangle` math in that function and track it down, but with the logging suppressed I think we can move **B** out to 2.0.
